packages/sysutils/busybox/scripts/getedid

```diff
@@ -0,0 +1,254 @@
+}
+
+
+check_content() {
+  # check if changes are already made to $file and exit if yes
+  if [ "$(grep "APPEND" $file)" = " APPEND boot=LABEL=System disk=LABEL=Storage ssh quiet" ]; then
+    echo "You alread did some changes to $file. Exiting."
+    exit 5
+  fi
+}
+
+
+# restore the backup done by the script and depending on the GPU
+restore() {
+  if [ "$gpu" = "intel" ]; then
+    check_file
+    mount_rw
+    if [ -f "$file".old ]; then
+      cp "$file".old "$file"  # cp because of keeping the backup...just in case
+    else
+      echo "You didn't created a custom EDID yet or you didn't use this script for this task."
+      echo "Therefore we can't ensure the script working properly."
+      echo "Exiting"
+      exit 3
+    fi
+  elif [ "$gpu" = "nvidia" ]; then
+    if [ -f /storage/.config/xorg.conf ]; then
+      mv /storage/.config/xorg.conf /storage/.config/xorg.le.backup
+    else
+      echo "You don't have a xorg.conf in your .config folder. Exiting"
+      exit 7
+    fi
+  fi
+  sys_reboot
+}
+
+
+intel() {
+  #check which output is connnected:
+  for i in /sys/class/drm/*; do
+    if [ "$(cat "$i"/status 2>/dev/null)" = "connected" ]; then
+      hdmi="$(echo "$i" | cut -d / -f 5 | sed 's/card[0-9]-//g')"
+    fi
+  done
+
+  #create edid
+  mkdir -p /storage/.config/firmware/edid
+  cat "/sys/class/drm/$hdmi/edid" > /storage/.config/firmware/edid/edid.bin
+
+  #create cpio archive
+  cd /storage
+  mkdir -p cpio/lib/firmware/edid
+  cp /storage/.config/firmware/edid/edid.bin /storage/cpio/lib/firmware/edid/
+  cd cpio/
+  find . -print | cpio -ov -H newc > /storage/edid.cpio
+
+  #remount /flash to rw
+  mount_rw
+  mv /storage/edid.cpio "$sys_path"
+
+  #check extlinux.conf and syslinux.cfg
+  check_file
+  check_content
+
+  # make a backup of $file
+  cp "$file" "$file".old
+
+  #add things to $file
+  sed -i "/ APPEND/s/$/ initrd=/edid.cpio drm_kms_helper.edid_firmware=$hdmi:edid/edid.bin video=$hdmi:D/" "$file"
+
+  #reboot
+  sys_reboot
+}
+
+
+nvidia() {
+  # check if xorg.conf already exists
+  if [ -f /storage/.config/xorg.conf ] || [ -f /storage/.config/xorg-nvidia.conf ]; then
+    echo "Existing /storage/.config/xorg.conf detected. Aborting!"
+    exit 6
+  fi
+
+
+  # set debug and restart Xorg
+  cp /etc/X11/xorg-nvidia.conf /storage/.config/xorg.conf
+  sed -i 's/"ModeDebug" "false"/"ModeDebug" "true"/g' /storage/.config/xorg.conf
+  systemctl restart xorg.service
+  
+
+  # get port
+  nv_port="$(cat /var/log/Xorg.0.log | grep -i -w connected | grep -i -o "dfp-[0-9]" | head -n 1)"
+  nvidia-xconfig --extract-edids-from-file=/var/log/Xorg.0.log --extract-edids-output-file=/storage/.config/edid.bin 1>/dev/null
+
+  # set mode debug back to false
+  sed -i 's/"ModeDebug" "true"/"ModeDebug" "false"/g' /storage/.config/xorg.conf
+  
+  # set port and uncomment lines
+  sed -i "s/#    Option         \"ConnectedMonitor\" \"DFP-0\"/    Option         \"ConnectedMonitor\" \"$nv_port\"/g" /storage/.config/xorg.conf
```

Comment:
The lines I change by using sed  can ofc be edited by the user before. But therefore I check if a xorg.conf already exists and exit if yes. If the user already did some changes to the xorg.conf, he's also able to do that a 2nd time ;) and then won't need that script